### PR TITLE
[EDU-3202] feat: Fingerprint integration guide

### PR DESCRIPTION
**Pull request by MarianaAguilera**
### Related issue:

[EDU-3202]

### Changes

- Adding new guides on how to install Fingerprint integration.

### Additional links
n/a.

[EDU-3202]: https://aziontech.atlassian.net/browse/EDU-3202?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+- Audio and video support
+- Canvas renderization
+- If the browser has access to device-level API, such as accelerometer and gyroscope
+- Manufacturer's data
+- Timezone
+- WebGL renderization
+
+Then, it creates a unique identifier containing all this information and sets it as a session cookie named `_azfp_sc`, as cookies are always sent in the requests made by the browser. The script checks if the session cookie already exists. If so, it won't create a new one..
+
+You need to embed the whole code in the browser. To do so, proceed as follows:
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:36]: validate process

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 2]
```diff
@@ -0,0 +1,162 @@
+---
+title: How to install the Fingerprint integration through Azion Marketplace
+description: The Azion Bot Protection analyzes incoming requests using edge functions on Edge Firewall, assigning a score based on rules and behaviors to detect and prevent malicious activities.
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:3]:
```suggestion
description: Use this integration to capture session data to better identify users and devices accessing your applications. 
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 3]
```diff
@@ -0,0 +1,162 @@
+---
+title: How to install the Fingerprint integration through Azion Marketplace
+description: Use this integration to capture session data to better identify users and devices accessing your applications. 
+meta_tags: bot, protection, cybersecurity, edge computing
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:4]:
```suggestion
meta_tags: protection, cybersecurity, edge computing
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+namespace: docs_use_case_bot_protection
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:5]:
```suggestion
namespace: docs_use_case_fingerprint
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+- Geoip Region Code: 
+- Header Accept: 
+- Header Accept-CH: 
+- Header Accept-Encoding: 
+- Header Accept-Language: 
+- Header User-Agent: 
+- Header X-Forwarded-For: 
+- Remote Address: 
+- SSL Cipher: 
+- SSL Version:
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:108]:
```suggestion
- Autonomous System Numbers (ASN)
- Cookie `_azfp_sc`, the cookie created by the [browser script](#embedding-the-script)
- Geoip City Code
- Geoip Country Code
- Geoip Region Code
- Header Accept
- Header Accept-CH
- Header Accept-Encoding
- Header Accept-Language
- Header User-Agent
- Header X-Forwarded-For
- Remote Address
- SSL Cipher
- SSL Version
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+Possible variables to be ignored in the creation of the fingerprint:
+
+- `cookie_fingerprint`: the cookie created by the [browser script](#embedding-the-script), named `_azfp_sc`
+- `header_accept`: the content types the client can understand. Example: `text/html`
+- `header_accept_ch`: the client hints headers a client should include in subsequent requests. Exmaple: `Viewport-Width, Width`
+- `header_accept_encoding`: the encoding algorithms a client supports. Example: `gzip`
+- `header_accept_language`: the language client prefers. Example: `"en-US"`
+- `header_user_agent`: the software and version the client uses. Example: `Chrome/51.0.2704.106`
+- `geoip_asn`: the ASN associated with the user's IP. Example: `52580`
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:126]:
```suggestion
- `geoip_asn`: the ASN associated with the device's IP. Example: `52580`
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 7]
```diff
@@ -0,0 +1,162 @@
+---
+title: How to install the Fingerprint integration through Azion Marketplace
+description: Use this integration to capture session data to better identify users and devices accessing your applications.
```
Comment (MarianaAguilera) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:3]:
```suggestion
description: Use this integration to capture session data to better identify devices accessing your applications. 
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+This integration includes two features:
+
+- A **browser-level script** (optional): a [JavaScript code](#embedding-the-script) that could be added to the web page. It captures data about the browser and sets it as a session cookie (named `_azfp_sc`), so this data is always sent by the browser. Even when the usage of this script is optional, it increases the precision of the integration.
+- An **edge function** (mandatory): a function to capture connection-related data as well as the "frontend fingerprint" (created by the browser script), when available, to create the session fingerprint. It'll then add this fingerprint as a request header (named `X-AZFP`).
+
+---
+
+## Embedding the script
+
+The browser script is a JavaScript code that runs in the web page you want to fingerprint, a process also know as JavaScript injection. It captures multiple data from the web browser:
```
Comment (gabriel-azion) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:20]:
```suggestion
The browser script is a JavaScript code that runs on the web page you want to fingerprint, a process also known as JavaScript injection. It captures multiple data from the web browser:
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+
+- Audio and video support
+- Canvas renderization
+- If the browser has access to device-level API, such as accelerometer and gyroscope
+- Manufacturer's data
+- Timezone
+- WebGL renderization
```
Comment (gabriel-azion) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:27]:
```suggestion
- WebGL rendering
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+
+:::note
+This is an optional step. However, using this script increases the precision of the **Fingerprint** integration. The code is a modularized script, in order to avoid collateral damage in any other scripts that might be running in the same web page.
+:::
+
+---
+
+## Getting the integration
+
+To install Azion **Fingerptint**:
```
Comment (gabriel-azion) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:48]:
```suggestion
To install Azion **Fingerprint**:
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+{
+  "fingerprint_ignore": []
+}
+```
+
+Possible variables to be ignored in the creation of the fingerprint:
+
+- `cookie_fingerprint`: the cookie created by the [browser script](#embedding-the-script), named `_azfp_sc`
+- `header_accept`: the content types the client can understand. Example: `text/html`
+- `header_accept_ch`: the client hints headers a client should include in subsequent requests. Exmaple: `Viewport-Width, Width`
```
Comment (gabriel-azion) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:119]:
```suggestion
- `header_accept_ch`: the client hints headers a client should include in subsequent requests. Example: `Viewport-Width, Width`
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+- `header_accept_encoding`: the encoding algorithms a client supports. Example: `gzip`
+- `header_accept_language`: the language client prefers. Example: `"en-US"`
```
Comment (gabriel-azion) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:121]:
```suggestion
- `header_accept_language`: the language the client prefers. Example: `"en-US"`
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+2. Click the **New Rule** button.
+3. Give an easy-to-remember name to the rule.
+   - You can add a description, but it's an optional step.
+4. Select a *criteria* to run and catch the domains that you want to run the integration on.
+    - Use this rule: 
+5. Below, select a *behavior* to the *criteria*. In this case, it'll be **Run Function**.
+6. Select the adequate function according to the name you gave it in the instantiate step.
+7. Click the **Save** button.
+
+Done. You now have you can collect session data by using Azion **Fingerptint**.
```
Comment (gabriel-azion) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:153]:
```suggestion
Done. You now have you can collect session data by using Azion **Fingerprint**.
```

File: src/content/docs/pt-br/pages/guias/marketplace/fingerprint/fingerprint.mdx [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+
+:::note
+Esse passo é opcional, mas o uso deste script aumenta a precisão da integração **Fingerprint**. O código é um script modularizado, para evitar danos colaterais em quaisquer outros scripts que possam estar sendo executados na mesma página da web.
+:::
+
+---
+
+## Obtenha a integração
+
+Para instalar **Fingerptint** da Azion:
```
Comment (gabriel-azion) [inline src/content/docs/pt-br/pages/guias/marketplace/fingerprint/fingerprint.mdx:48]:
```suggestion
Para instalar o **Fingerprint** da Azion:
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+Done. Now you've instantiated the rule for your function and have access to edge functions on your edge firewall.
+
+### Setting up the function
+
+While still on the **Edge Firewall** page:
+
+1. Select the **Functions** tab.
+2. Click the **Add Function** button.
+3. Give an easy-to-remember name to your instance.
+4. On the dropdown menu, select the **Fingerptint** function.
```
Comment (gabriel-azion) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:84]:
```suggestion
4. On the dropdown menu, select the **Fingerprint** function.
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+meta_tags: protection, cybersecurity, edge computing
+namespace: docs_use_case_fingerprint
+permalink: /documentation/products/guides/fingerprint/
+---
+
+Azion **Fingerprint** enables you to capture devices and network data to better identify devices accessing your applications. It can be classified as a *Session Fingerprinting* integration and uses the collected data to generate a unique identifier and add it as a request header.
+
+This integration includes two features:
+
+- A **browser-level script** (optional): a [JavaScript code](#embedding-the-script) that could be added to the web page. It captures data about the browser and sets it as a session cookie (named `_azfp_sc`), so this data is always sent by the browser. Even when the usage of this script is optional, it increases the precision of the integration.
```
Comment (bru-andrade) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:13]:
```suggestion
- A **browser-level script** (optional): a [JavaScript code](#embedding-the-script) that could be added to the web page. It captures data about the browser and sets it as a session cookie named `_azfp_sc`, so this data is always sent by the browser. Even when the usage of this script is optional, it increases the precision of the integration.
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 17]
```diff
@@ -0,0 +1,159 @@
...
+While still on the **Edge Firewall** page:
+
+1. Select the **Functions** tab.
+2. Click the **Add Function** button.
+3. Give an easy-to-remember name to your instance.
+4. On the dropdown menu, select the **Fingerptint** function.
+
+This action will load the function, showing a form with the function code and, just above it, two tabs: **Code** and **Args**.
+
+By clicking the Code tab, you'll be able to navigate through the source code but not to change it.
```
Comment (hannah-sk) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:88]:
```suggestion
By clicking the **Code** tab, you'll be able to navigate through the source code but not to change it.
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 18]
```diff
@@ -0,0 +1,159 @@
...
+- Cookie `_azfp_sc`
+- Geoip City Code
+- Geoip Country Code
+- Geoip Region Code
+- Header Accept
+- Header Accept-CH
+- Header Accept-Encoding
+- Header Accept-Language
+- Header User-Agent
+- Header X-Forwarded-For
```
Comment (bru-andrade) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:102]:
```suggestion
- Header `Accept`
- Header `Accept-CH`
- Header `Accept-Encoding`
- Header `Accept-Language`
- Header `User-Agent`
- Header `X-Forwarded-For`
```

File: src/content/docs/pt-br/pages/guias/marketplace/fingerprint/fingerprint.mdx [hunk 19]
```diff
@@ -0,0 +1,159 @@
...
+## Embede o script
+
+O script do navegador é um código JavaScript que é executado na página da web que você deseja identificar, um processo também conhecido como injeção de JavaScript. Ele captura vários dados do navegador web:
+
+- Suporte a áudio e vídeo
+- Renderização de Canvas
+- Se o navegador tem acesso à API de nível de dispositivo, como acelerômetro e giroscópio
+- Dados do fabricante
+- Fuso horário
+- Renderização WebGL.
```
Comment (hannah-sk) [inline src/content/docs/pt-br/pages/guias/marketplace/fingerprint/fingerprint.mdx:27]:
```suggestion
- Renderização WebGL
```

File: src/content/docs/pt-br/pages/guias/marketplace/fingerprint/fingerprint.mdx [hunk 20]
```diff
@@ -0,0 +1,159 @@
...
+
+Em seguida, ele cria um identificador único contendo todas essas informações e o define como um cookie de sessão chamado `_azfp_sc`, pois os cookies são sempre enviados nas requisições feitas pelo navegador. O script verifica se o cookie de sessão já existe. Se sim, ele não criará um novo.
+
+Azion fornece um exemplo de código para fazer a injeção de JavaScript. 
+
+1. Você pode encontrar o código [nesse arquivo](https://mal2u8n8zk.map.azionedge.net/274addd29ac1edb2b200a297b98581cd88465e2ee36f18cfdbdc063402063fe5.js)
+2. Logo, adicioná-lo no seu HTML fonte. Exemplo:
```
Comment (hannah-sk) [inline src/content/docs/pt-br/pages/guias/marketplace/fingerprint/fingerprint.mdx:34]:
```suggestion
2. Em seguida, adicioná-lo no seu HTML fonte. Exemplo:
```

File: src/content/docs/pt-br/pages/guias/marketplace/fingerprint/fingerprint.mdx [hunk 21]
```diff
@@ -0,0 +1,159 @@
...
+Ainda na página **Edge Firewall**:
+
+1. Selecione a aba **Functions**.
+2. Clique no botão **Add Function**.
+3. Dê um nome fácil de lembrar para sua instância.
+4. No menu suspenso, selecione a função **Fingerprint**.
+
+Esta ação carregará a função, mostrando um formulário com o código da função e, logo acima dele, duas abas: **Code** e **Args**.
+
+Ao clicar na aba Code, você poderá navegar pelo código-fonte, mas não poderá alterá-lo.
```
Comment (hannah-sk) [inline src/content/docs/pt-br/pages/guias/marketplace/fingerprint/fingerprint.mdx:88]:
```suggestion
Ao clicar na aba **Code**, você poderá navegar pelo código-fonte, mas não poderá alterá-lo.
```

File: src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx [hunk 22]
```diff
@@ -0,0 +1,159 @@
...
+- `geoip_asn`: the ASN associated with the device's IP. Example: `52580`
+- `geoip_city`: the city associated with the device's IP. Example: `"Brasilia"`
+- `geoip_country_code`: the country code associated with the device's IP. Example: `"BR"`
+- `geoip_region`: the region or state associated with the device's IP. Example: `"DF"`
+- `remote_addr`: the device's remote IP address. Example: `000.0.0.0`
+- `ssl_cipher`: the SSL cipher used in the device's connection. Example: `"ECDHE-RSA-AES128-GCM-SHA256"`
+- `ssl_protocol`: the SSL protocol used in the device's connection. Example: `"TLSv1.2"`
+
+:::note
+Any variable as "null" or blank will be still a valid value.
```
Comment (bru-andrade) [inline src/content/docs/en/pages/guides/marketplace/integrations/fingerprint/fingerprint.mdx:132]:
```suggestion
Any variable as `null` or blank will be still a valid value.
```

**Review by hannah-sk — APPROVED** [review]

File: src/content/docs/pt-br/pages/guias/marketplace/fingerprint/fingerprint.mdx [hunk 23]
```diff
@@ -0,0 +1,159 @@
...
+Para finalizar, você precisa configurar o **Rules Engine** para definir o *criteria* (critério) e o *behavior* (comportamento) para executar a função.
+
+Ainda na página do **Edge Firewall**:
+
+1. Selecione a aba **Rules Engine**.
+2. Clique no botão **New Rule**.
+3. Dê um nome fácil de lembrar para a regra.
+   - Você pode adicionar uma descrição, mas é um passo opcional.
+4. Selecione um  *criteria* para executar e capturar os domínios em que você deseja executar a integração.
+    - Use esta regra:
```
Comment (MarianaAguilera) [inline src/content/docs/pt-br/pages/guias/marketplace/fingerprint/fingerprint.mdx:148]:
```suggestion
    - Defina uma regra, de acordo com suas necessidades, para gerar o fingerprint.
```

**Pull request merged**